src/engine/source/queue/interface/queue/iqueue.hpp

```diff
@@ -0,0 +1,78 @@
+#ifndef _QUEUE_IQUEUE_HPP
+#define _QUEUE_IQUEUE_HPP
+
+
+
+namespace base::queue
+{
+
+/**
+ * @brief An interface for a blocking concurrent queue
+ *
+ * This interface defines the common operations for a blocking concurrent queue.
+ * Implementations should provide methods for pushing, waiting for a pop operation,
+ *
+ * @tparam T The type of elements to be stored in the queue.
+ */
+template<typename T>
+class iQueue
+{
+public:
+    /**
+     * @brief Destructor for the interface.
+     */
+    virtual ~iQueue() = default;
+
+    /**
+     * @brief Push an element into the queue if fails to push, flush element to file.
+     *
+     * @param element The element to push into the queue
+     */
+    virtual void push(T&& element) = 0;
+
+    /**
+     * @brief Try to push an element into the queue
+     *
+     * @param element
+     * @return true if the element was pushed successfully, false otherwise.
+     */
+    virtual bool tryPush(const T& element) = 0;
```

Comment:
If we succeed in pushing we copy the element, should we use a non const reference? It the function fails the element is not copied or moved.